mathesar_ui/src/component-library/toast/ToastController.ts

```diff
@@ -0,0 +1,230 @@
+import {
+  faCheck,
+  faExclamationTriangle,
+  faSpinner,
+} from '@fortawesome/free-solid-svg-icons';
+import type { IconDefinition } from '@fortawesome/free-solid-svg-icons';
+import type { SvelteComponent } from 'svelte';
+import { linear } from 'svelte/easing';
+import type { Writable, Readable } from 'svelte/store';
+import { writable, derived } from 'svelte/store';
+import type { PauseableTweened } from '../common/utils/pauseableTweened';
+import { pauseableTweened } from '../common/utils/pauseableTweened';
+import type { IconFlip, IconRotate } from '../icon/Icon.d';
```

Comment:
The imports need some cleanup with respect to using alias. We need to consistently use it for the component-library and for any relative path that backtracks to a parent directory.

Reply:
For line 12, is this what you had in-mind?

```ts
import { pauseableTweened } from '@mathesar-component-library';
```

With that import I get the following linting error

> Dependency cycle detected

What's the proper way to resolve this error?

Reply:
This error would only occur when there is a circular dependency link between two files, either directly or indirectly. eg., File A depends on File B, which inturn depends on File A.

In this case, I assume it's because there is a chain cause by one of the imports, probably caused by using the:
```
import { ... } from '@mathesar-component-library'
```
alias somewhere. This is an alias to an index.ts file which inturn imports all the components and utilities from within the library. So using this alias within the library is certain cause for this error which would also result in build failure.

Instead, all the files within the library folder should import each other by using the:
```
import File from '@mathesar-component-library-dir'
```
alias.

The `mathesar-component-library-dir` alias is only for use within the component library. `mathesar-component-library` alias is only for use outside the component library.

Reply:
@pavish I put some changes in the `758_toast__imports` branch that I think are in line with what you're requesting. However, with those changes I'm unable to start Storybook. Can you take a look and help me resolve the errors? 

It's also worth noting that I see lots of files inside `src/component-library` which contain imports using the `@mathesar-component-library` alias, thereby creating a circular import. It seems to me that the linter just doesn't notice because those imports are in `.svelte` files.

```
grep -Pr '@mathesar-component-library(?!-)' src/component-library
```

Reply:
@seancolsen This commit https://github.com/centerofci/mathesar/pull/821/commits/2dc2113b688ebec0504f1c7b3c2a57f93d0a02de should fix the alias resolution.

Storybook should now work when we import with `@mathesar-component-library-dir`

Reply:
I cherry-picked that commit onto this branch now as b65f6053f9f56ef74323df47be69c6ec14478444, which I think should resolve this issue